tezos/context/src/working_tree/working_tree.rs

```diff
@@ -87,8 +88,8 @@ use super::{
 
 pub struct PostCommitData {
     pub commit_ref: ObjectReference,
-    pub batch: Vec<(HashId, Arc<[u8]>)>,
-    pub reused: Vec<HashId>,
+    pub batch: ChunkedVec<(HashId, Arc<[u8]>)>,
```

Comment:
General question, as long as the GC runs in parallel we cannot get rid of all the `Arc`s right? I think we discussed before but would it be possible to use coarser locking? having locks around big chunks of data instead of single values, I think that could save a lot of memory

Reply:
Yes, we need to `Arc` because of the GC. I will use coarser locking in my next PR https://github.com/tezedge/tezedge/pull/940